Admin/FragmentAdmin.php

```diff
@@ -0,0 +1,273 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\ArticleBundle\Admin;
+
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\AdminBundle\Route\RouteCollection;
+use Sonata\AdminBundle\Show\ShowMapper;
+use Sonata\ArticleBundle\FragmentService\FragmentServiceInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author  Hugo Briand <briand@ekino.com>
+ */
+final class FragmentAdmin extends AbstractAdmin
+{
+    /**
+     * @var FragmentServiceInterface[]
+     */
+    private $fragmentServices = array();
+
+    /**
+     * @var array
+     */
+    private $settings = array();
+
+    /**
+     * @param array $fragmentServices
+     */
+    final public function setFragmentServices(array $fragmentServices)
+    {
+        $this->fragmentServices = $fragmentServices;
+    }
+
+    /**
+     * @return FragmentServiceInterface[]
+     */
+    public function getFragmentServices()
+    {
+        return $this->fragmentServices;
+    }
+
+    /**
+     * @param array $settings
+     */
+    final public function setSettings(array $settings)
+    {
+        $this->settings = $settings;
+    }
+
+    /**
+     * @return array
+     */
+    public function getSettings()
+    {
+        return $this->settings;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTemplate($name)
+    {
+        if ('edit' === $name) {
+            if ($this->getRequest()->get('type')) {
+                return $this->getService($this->getRequest()->get('type'))->getEditTemplate();
+            }
+
+            return 'SonataArticleBundle:FragmentAdmin:form.html.twig';
+        }
+
+        return parent::getTemplate($name);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFormBuilder()
+    {
+        $this->formOptions['data_class'] = $this->getClass();
+        $this->formOptions['csrf_protection'] = false;
+
+        // Our form will be built without knowing of its parent (article), so we need to specify it in its name
```

Comment:
Why will it be built without knowing of its parent? And why do we need to specify it in its name? To reattach it or something?